kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -273,16 +269,21 @@ private static ProduceResponse toProduceResponse(
         .build();
   }
 
-  private void recordResponseMetrics(long latency) {
+  private void recordResponseMetrics(Duration latency) {
     producerMetrics.get().recordResponse();
     producerMetrics.get().recordRequestLatency(latency);
   }
 
-  private void recordErrorMetrics(long latency) {
+  private void recordErrorMetrics(Duration latency) {
     producerMetrics.get().recordError();
     producerMetrics.get().recordRequestLatency(latency);
   }
 
+  private void recordRateLimitedMetrics(Duration latency) {
+    producerMetrics.get().recordRateLimited();
+    producerMetrics.get().recordRequestLatency(latency);
```

Comment:
Do we want to include rate-limited requests in the latency metric?

Reply:
I think if we did want a latency metric for rate limited requests, then it should be a separate metric, in the same way you have done for the rate limited count.

I guess the question is, do we think we are going to suffer from increased latency in the rate limiting part of the flow.

I think we might, if it's really busy, so we probably do care about this metric too.

Reply:
I commented out-of-thread on this - I don't think the measurement of the latency of the rate-limiting functionality is interesting. It doesn't take into account any sort of pipelining or queuing effects that might be telling.

Reply:
Done.